File: DSA/Ayush_Mishra_590014844/Day_38_Question_3.c
#include <stdio.h>
#include <string.h>
#define MAX 100

int main() {
    char expr[MAX];
    char stack[MAX];
    int top = -1;

    printf("Enter expression: ");
    scanf("%s", expr);

    for(int i = 0; expr[i]; i++) {
        if(expr[i] == ')') {
            int flag = 1;
            while(stack[top] != '(') {
                if(stack[top] == '+' || stack[top] == '-' || stack[top] == '*' || stack[top] == '/') {
                    flag = 0;
                }
                top--;
            }
            top--;
            if(flag) {
                printf("Yes\n");
                return 0;
            }
        } else {
            stack[++top] = expr[i];
        }
    }

    printf("No\n");
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_38_Question_3.py
def has_redundant(expr):
    stack = []
    for ch in expr:
        if ch == ')':
            if stack.pop() == '(':
                return True
            while stack.pop() != '(':
                continue
        else:
            stack.append(ch)
    return False

expr = input("Enter expression: ")
print("Yes" if has_redundant(expr) else "No")
